src/app/games/diam/DiamState.ts

```diff
@@ -0,0 +1,63 @@
+import { GameStateWithTable } from 'src/app/jscaip/GameStateWithTable';
+import { Table } from 'src/app/utils/ArrayUtils';
+import { assert } from 'src/app/utils/utils';
+import { DiamPiece } from './DiamPiece';
+
+export class DiamState extends GameStateWithTable<DiamPiece> {
+    public static WIDTH: number = 8;
+
+    public static HEIGHT: number = 4;
+
+    public static getInitialState(): DiamState {
+        const _: DiamPiece = DiamPiece.EMPTY;
+        const board: Table<DiamPiece> = [
+            [_, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _],
+        ];
+        return new DiamState(board, [4, 4, 4, 4], 0);
+    }
+    public static fromRepresentation(representation: DiamPiece[][], turn: number): DiamState {
+        const board: Table<DiamPiece> = representation.reverse();
```

Comment:
un .reverse() à la fin de chaque board the tuto et test eu peut être suffit :o :O :⭕

Reply:
Même  si ça c'est plus beau ?

Reply:
ici ça simplifie le calcul des pièces restantes en plus de juste le .reverse, donc plus utile ?